lib/rest-api.php

```diff
@@ -30,3 +30,49 @@ function gutenberg_register_edit_site_export_controller_endpoints() {
 }
 
 add_action( 'rest_api_init', 'gutenberg_register_edit_site_export_controller_endpoints' );
+
+if ( ! function_exists( 'gutenberg_register_wp_rest_post_types_controller_fields' ) ) {
+	/**
+	 * Adds `template` and `template_lock` fields to WP_REST_Post_Types_Controller class.
+	 */
+	function gutenberg_register_wp_rest_post_types_controller_fields() {
+		register_rest_field(
+			'type',
+			'template',
+			array(
+				'get_callback' => function ( $item ) {
+					$post_type = get_post_type_object( $item['slug'] );
+					if ( ! empty( $post_type ) && ! empty( $post_type->template ) ) {
+						return $post_type->template;
+					}
+				},
+				'schema'       => array(
+					'type'        => 'array',
+					'description' => __( 'The template associated with the post type.', 'gutenberg' ),
```

Comment:
Should the phrasing reflect that the template is optional and refers to _block_ templates as opposed to a `wp_template`? For example:

```diff
-The template associated with the post type.
+The block template associated with the post type, if it exists.
```